src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java

```diff
@@ -1898,6 +1774,146 @@ public XMPPError onRemoveSource(JingleSession sourceJingleSession,
             sourceJingleSession, sourcesToRemove, sourceGroupsToRemove, true);
     }
 
+    /**
+     * Updates the RTP description, transport and propagates sources and source
+     * groups of a participant that sends the session-accept or transport-accept
+     * Jingle IQs.
+     *
+     * @param jingleSession
+     * @param contents
+     * @return
+     */
+    private XMPPError updateParticipant(
+        JingleSession jingleSession,
+        List<ContentPacketExtension> contents)
+    {
+        Participant participant
+            = findParticipantForJingleSession(jingleSession);
+        Jid participantJid = jingleSession.getAddress();
+
+        if (participant == null)
+        {
+            String errorMsg
+                = "No participant found for: " + participantJid;
+            logger.warn(errorMsg);
+            return XMPPError.from(XMPPError.Condition.item_not_found,
+                errorMsg).build();
+        }
+
+        // XXX We will be acting on the received session-accept bellow.
+        // Unfortunately, we may have not received an acknowledgment of our
+        // session-initiate yet and whatever we do bellow will be torn down when
+        // the acknowledgement timeout occurs later on. Since we will have
+        // acted on the session-accept by the time the acknowledgement timeout
+        // occurs, we may as well ignore the timeout.
+        jingleSession.setAccepted(true);
+
+
+        // Extract and store various session information in the Participant
+        participant.setRTPDescription(contents);
+        participant.addTransportFromJingle(contents);
+
+        MediaSourceMap sourcesAdvertised
+            = MediaSourceMap.getSourcesFromContent(contents);
+        MediaSourceGroupMap sourceGroupsAdvertised
+            = MediaSourceGroupMap.getSourceGroupsForContents(contents);
+        if (sourcesAdvertised.isEmpty()
+            && globalConfig.injectSsrcForRecvOnlyEndpoints)
+        {
+            // We inject an SSRC in order to insure that the participant has
+            // at least one SSRC advertised. Otherwise, non-local bridges in the
+            // conference will not be aware of the participant. We intentionally
+            // use a negative value, because this is an invalid SSRC and will
```

Comment:
Do we? It doesn't look like it. (I know this is an old, moved comment...)

Reply:
The code seems to, indeed, inject an ssrc in the advertised sources. The comment indicates that this is necessary for octo operation. It feels risky and unnecessary to change this behavior without deeper understanding or without the original authors "blessing"?

Reply:
Sorry, the "do we" comment was in reference to the "negative value" - I don't think this part is right?

Reply:
@bgrozev shall we remove the reference to the "negative value" or update the code to reflect the behavior described in the comment?

Reply:
I don't remember why I wrote that, but my best guess is that I tried using a negative value, something broke, and then I changed the code but forgot to fix my comment. Let's keep the code and update the comment.